packages/components/src/internal/components/assay/AssayImportPanels.tsx

```diff
@@ -294,14 +297,18 @@ class AssayImportPanelsBody extends Component<Props, State> {
     }
 
     onGetQueryDetailsComplete = (): void => {
-        const { assayDefinition, location } = this.props;
+        const { assayDefinition, location, loadJobSamples, loadSelections } = this.props;
         const sampleColumnData = assayDefinition.getSampleColumn();
 
         if (sampleColumnData && location) {
             // If the assay has a sample column look up at Batch, Run, or Result level then we want to retrieve
             // the currently selected samples so we can pre-populate the fields in the wizard with the selected
             // samples.
-            this.props.loadSelections(location, sampleColumnData.column).then(samples => {
+            // Additionally, we want to pre-populate the fields in the wizard with samples if the job has samples
+            // associated with it, and the assay has a sample column.
+
+            const loadGridWithSamplesFn = (location.query?.workflowJobId && isSampleManagerEnabled()) ? loadJobSamples : loadSelections;
+            loadGridWithSamplesFn(location, sampleColumnData.column).then(samples => {
```

Comment:
As discussed I think this logical check against the location can be moved down into the method (and out of the component). This will align with what is already being done by `getSelections()`. Something like:

```ts
export function loadSelectedSamples(location: Location, sampleColumn: QueryColumn): Promise<OrderedMap<any, any>> {
    // If the "workflowJobId" URL parameter is specified, then fetch the samples associated with the workflow job.
    if (location?.query?.workflowJobId) {
        return fetchSamples(
            SchemaQuery.create('sampleManagement', 'inputSamples'),
            sampleColumn,
            [
                Filter.create('ApplicationType', 'ExperimentRun'),
                Filter.create('ApplicationRun', location.query.workflowJobId),
            ],
            'Name',
            'SampleId'
        );
    }

    // Otherwise, load the samples from the selection.
    return getSelection(location).then(selection => {
        if (selection.resolved && selection.schemaQuery && selection.selected.length) {
            return fetchSamples(
                selection.schemaQuery,
                sampleColumn,
                [Filter.create('RowId', selection.selected, Filter.Types.IN)],
                sampleColumn.lookup.displayColumn,
                sampleColumn.lookup.keyColumn
            );
        }

        return OrderedMap();
    });
}
```